File: frontend/src/types/team.ts
export interface TeamDriver {
  driver_name: string;
  name_acronym: string;
  driver_number: number | null;
  points: number;
  driver_position: number | null;
  headshot_url: string | null;
}

export interface Team {
  position: number;
  constructor_id: string;
  constructor_name: string;
  team_colour: string;
  points: number;
  wins: number;
  drivers: TeamDriver[];
  // Detail fields from team_details seed table
  full_name: string | null;
  base_city: string | null;
  base_country: string | null;
  founded_year: number | null;
  engine_supplier: string | null;
  team_principal: string | null;
  world_championships: number | null;
}

File: backend/app/models/team.py
from pydantic import BaseModel


class TeamDriver(BaseModel):
    driver_name: str
    name_acronym: str
    driver_number: int | None = None
    points: float = 0.0
    driver_position: int | None = None
    headshot_url: str | None = None


class Team(BaseModel):
    position: int
    constructor_id: str
    constructor_name: str
    team_colour: str
    points: float
    wins: int
    drivers: list[TeamDriver]
    # Detail fields from team_details seed table
    full_name: str | None = None
    base_city: str | None = None
    base_country: str | None = None
    founded_year: int | None = None
    engine_supplier: str | None = None
    team_principal: str | None = None
    world_championships: int | None = None


class TeamsResponse(BaseModel):
    year: int
    round: int
    teams: list[Team]
